lib/matplotlib/axes/_axes.py

```diff
@@ -4559,6 +4559,11 @@ def scatter(self, x, y, s=None, c=None, marker=None, cmap=None, norm=None,
             collection.set_cmap(cmap)
             collection.set_norm(norm)
             collection._scale_norm(norm, vmin, vmax)
+        elif any({cmap, norm, vmin, vmax}):
+            _api.warn_external(
+                    "kwargs cmap, norm, vmin, vmax"
+                    "are being overridden by the default color mapping,"
+                    "try passing an empty c parameter")
```

Comment:
Two problems here:

1. These strings will be concatenated without spaces, so your message will look like "vmaxare being" and "mapping,try passing". Just add spaces at the end or beginning of each string.
2. This current message is appropriate for my scenario in #23487 but more generally I think we should warn any time someone tries to pass one of these parameters without `c`. So maybe something generic like `"'c' not given, 'cmap', 'norm', 'vmin', and 'vmax' will be ignored"`. Even better if we can specify which of cmap/norm/vmin/vmax is being used/ignored.